src/app-layer-parser.c

```diff
@@ -655,7 +662,7 @@ uint64_t AppLayerTransactionGetActiveLogOnly(Flow *f, uint8_t flags)
     AppLayerParserProtoCtx *p = &alp_ctx.ctxs[f->protomap][f->alproto];
 
     if (p->logger == TRUE) {
```

Comment:
I think it's important that we now also track whether we have loggers per direction. If we have a logger for one direction, is the log_id for the other dir also updated somehow? Otherwise this func will always return tx id 0